### PR TITLE
imx8: imx8mp: create pd24.1.2 mainline documentation

### PR DESCRIPTION
**Pull request by Varakala02**
Created PD24.1.2 minor mainline release documentation with the release information and compatile BSP information.

Added pd24.1.2.rst to the imx8mp.rst index file and to the pdf output conf.py file.

### CODE REVIEW

File: source/bsp/imx8/imx8mp/pd24.1.2.rst [hunk 1]
```diff
@@ -0,0 +1,1363 @@
...
+   on the |sbc| can be found here:
+   https://github.com/phytec/linux-phytec/blob/v6.6.21-phy1/arch/arm64/boot/dts/freescale/imx8mp-phyboard-pollux-rdk.dts#L106
+.. |pollux-fan-note| replace::
+   Starting with BSP-Yocto-i.MX8MP-PD22.1.1 we have to switch from PWM fan
+   to GPIO fan due to availability. The PWM fan will not be supported
+   anymore and will not function with the new release.
+
+.. |ref-serial| replace:: :ref:`X2 <imx8mp-mainline-head-components>`
+.. |ref-jp3| replace:: :ref:`JP3 <imx8mp-mainline-head-components>`
+.. |ref-jp4| replace:: :ref:`JP4 <imx8mp-mainline-head-components>`
```
Comment (ymoog) [inline source/bsp/imx8/imx8mp/pd24.1.2.rst:102]: The references need to be updated. This needs to be done for every :ref: 

<imx8mp-pd24.1.2-components>

File: source/bsp/imx8/imx8mp/pd24.1.2.rst [hunk 2]
```diff
@@ -0,0 +1,1363 @@
...
+.. |mcore-zephyr-docs| replace:: https://docs.zephyrproject.org/latest/boards/phytec/mimx8mp_phyboard_pollux/doc/index.html
+.. |mcore-jtag-dev| replace:: MIMX8ML8_M7
+.. |mcore-sdk-version| replace:: 2.13.0
+
++-----------------------+----------------------+
+| |doc-id| |soc| BSP    |                      |
+| ManualHead            |                      |
++-----------------------+----------------------+
+| Document Title        | |doc-id| |soc| BSP   |
+|                       | Mainline Manual Head |
```
Comment (tremmet) [inline source/bsp/imx8/imx8mp/pd24.1.2.rst:116]: Remove the 'Head' here.

File: source/bsp/imx8/imx8mp/pd24.1.2.rst [hunk 3]
```diff
@@ -0,0 +1,1363 @@
...
+| Document Type         | BSP Manual           |
++-----------------------+----------------------+
+| Article Number        | |doc-id|             |
++-----------------------+----------------------+
+| Yocto Manual          | Scarthgap            |
++-----------------------+----------------------+
+| Release Date          | 2024/06/26           |
++-----------------------+----------------------+
+| Is Branch of          | |doc-id| |soc| BSP   |
+|                       | Mainline Manual Head |
```
Comment (tremmet) [inline source/bsp/imx8/imx8mp/pd24.1.2.rst:127]: Same here.

**Review by tremmet — CHANGES_REQUESTED** [review]

File: source/bsp/imx8/imx8mp/pd24.1.2.rst (hunk 2 above)
Comment (Varakala02) [reply source/bsp/imx8/imx8mp/pd24.1.2.rst]: Thanks updated

**Comment by tremmet** [comment]
@jonas-rem  can you create a tag for the PD24.1.2 mainline i.MX8MP Release after merging this?

**Review by tremmet — APPROVED** [review]

**Review by ymoog — APPROVED** [review]

**Review by ymoog — APPROVED** [review]

**Pull request merged**